### PR TITLE
Create order repository

### PR DESCRIPTION
**Pull request by vnayakg**

### CODE REVIEW

File: src/main/kotlin/com/esops/repository/OrderRepository.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import com.esops.entity.OrderType
+import com.esops.entity.SellOrderComparator
+import jakarta.inject.Singleton
+import java.math.BigInteger
+import java.util.*
+
+@Singleton
+class OrderRepository {
+
+    private var orders = HashMap<String, HashMap<Long, Order>>()
```
Comment (anirudhr95) [inline src/main/kotlin/com/esops/repository/OrderRepository.kt:14]: No way orders is of this type. There is a better way to do this

File: src/main/kotlin/com/esops/repository/OrderRepository.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    fun getOrderByUsername(username: String): List<Order> {
+        return orders[username]!!.values.toList()
+    }
+
+    fun addOrder(username: String, order: Order) {
+        orders[username]?.set(orderIDCounter, order)
+    }
+
+    fun getOrderQueue(type: OrderType): PriorityQueue<Order> {
+        if(type == OrderType.BUY)   return buyOrderQueue
```
Comment (anirudhr95) [inline src/main/kotlin/com/esops/repository/OrderRepository.kt]: this function should be an expression

File: src/main/kotlin/com/esops/repository/OrderRepository.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    fun cleanQueue() {
+        buyOrderQueue.removeIf { it.remainingQuantity == BigInteger("0") }
+        sellOrderQueue.removeIf { it.remainingQuantity == BigInteger("0") }
+    }
+
+    fun clearOrders() {
+        orders = HashMap()
+        orderIDCounter = 0
+        buyOrderQueue.clear()
```
Comment (anirudhr95) [inline src/main/kotlin/com/esops/repository/OrderRepository.kt:69]: clear? What is clean orders - clearly you are not just `cleaningOrders`

File: src/main/kotlin/com/esops/service/OrderService.kt [hunk 4]
```diff
@@ -3,25 +3,24 @@ package com.esops.service
...
-    private var orders = HashMap<String, HashMap<Long, Order>>()
-    private var buyOrderQueue = PriorityQueue(BuyOrderComparator)
-    private var sellOrderQueue = PriorityQueue(SellOrderComparator)
-    private var orderIDCounter: Long = 0
 
     fun placeOrder(username: String, addOrderRequestBody: AddOrderRequestBody): Order {
         userService.checkOrderPlacement(username, addOrderRequestBody)
-        orderIDCounter++
+        orderRepository.incrementOrderIDCounter()
         val user = this.userService.getUser(username)
```
Comment (anirudhr95) [inline src/main/kotlin/com/esops/service/OrderService.kt:21]: NONE OF THIS SHOULD HAPPEN IN SERVICE. If anything is store related, you have a repo, handle it there

File: src/main/kotlin/com/esops/service/OrderService.kt [hunk 5]
```diff
@@ -103,30 +102,25 @@ class OrderService(private var userService: UserService,
...
             }
         }
-        buyOrderQueue = clone
-        cleanQueue()
-    }
-
-    private fun cleanQueue() {
-        buyOrderQueue.removeIf { it.remainingQuantity == BigInteger("0") }
-        sellOrderQueue.removeIf { it.remainingQuantity == BigInteger("0") }
+        orderRepository.setOrderQueue(OrderType.BUY, clone)
```
Comment (anirudhr95) [inline src/main/kotlin/com/esops/service/OrderService.kt]: what is clone

File: src/main/kotlin/com/esops/service/OrderService.kt [hunk 6]
```diff
@@ -198,19 +193,8 @@ class OrderService(private var userService: UserService,
...
-    private fun initializeOrderMapForUser(username: String) {
-        orders[username] = HashMap()
-    }
-
-    fun clearOrders() {
-        orders = HashMap()
-        orderIDCounter = 0
-        buyOrderQueue.clear()
-        sellOrderQueue.clear()
+        orderRepository.initializeOrderMapIfEmpty(username)
```
Comment (anirudhr95) [inline src/main/kotlin/com/esops/service/OrderService.kt]: same. This is not a service responsibility

File: src/test/kotlin/com/esops/repository/OrderRepositoryTest.kt [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        val orderID = orderRepository.getOrderIDCounter()
+
+        orderRepository.incrementOrderIDCounter()
+
+        assertEquals(orderID + 1, orderRepository.getOrderIDCounter())
+    }
+
+    @Test
+    fun `should initialize empty order map for user`() {
+        orderRepository.initializeOrderMapIfEmpty("ramesh")
```
Comment (anirudhr95) [inline src/test/kotlin/com/esops/repository/OrderRepositoryTest.kt]: 🤨

File: src/main/kotlin/com/esops/repository/OrderRepository.kt (hunk 2 above)
Comment (vnayakg) [reply src/main/kotlin/com/esops/repository/OrderRepository.kt]: Created separate functions for buy order queue and sell order queue

File: src/main/kotlin/com/esops/repository/OrderRepository.kt (hunk 3 above)
Comment (vnayakg) [reply src/main/kotlin/com/esops/repository/OrderRepository.kt:69]: Only using in controller tests, unit tests changed

File: src/main/kotlin/com/esops/service/OrderService.kt (hunk 4 above)
Comment (vnayakg) [reply src/main/kotlin/com/esops/service/OrderService.kt:21]: MovedorderID increment part in repository

File: src/main/kotlin/com/esops/service/OrderService.kt (hunk 5 above)
Comment (vnayakg) [reply src/main/kotlin/com/esops/service/OrderService.kt]: rename clone to appropriate order queues

File: src/main/kotlin/com/esops/service/OrderService.kt (hunk 6 above)
Comment (vnayakg) [reply src/main/kotlin/com/esops/service/OrderService.kt]: moved to repository

File: src/test/kotlin/com/esops/repository/OrderRepositoryTest.kt (hunk 7 above)
Comment (vnayakg) [reply src/test/kotlin/com/esops/repository/OrderRepositoryTest.kt]: also same as above, moved to repository

**Pull request merged**